### PR TITLE
DB Reading integration test

### PR DESCRIPTION
**Pull request by pikonha**
# DB Reading Integration Test

## All Submissions:

* [X] Have you followed the guidelines in our Contributing document?
* [X] Have you checked to ensure there aren't other open [Pull Requests](../../../pulls) for the same update/change?

## Description

It creates all the components required to run the full flow of the database resolution and then tests some of the main scenarios.

## Changes
- [X] New feature implementation
- [ ] Bug fix
- [ ] Code refactoring
- [ ] Documentation update
- [ ] Other (please specify)

### Changes to Core Features:

* [X] Have you added an explanation of what your changes do and why you'd like us to include them?
* [X] Have you written new tests for your core changes, as applicable?
* [X] Have you successfully run tests with your changes locally?

### CODE REVIEW

File: packages/client/package.json [hunk 1]
```diff
@@ -2,12 +2,12 @@
   "name": "@blockful/client",
   "packageManager": "yarn@4.1.0",
   "scripts": {
-    "test": "vitest run",
-    "test:watch": "vitest",
+    "test:db:e2e": "hardhat test ./test/db.e2e.spec.ts --network localhost",
```
Comment (pikonha) [inline packages/client/package.json:5]: this is enable the creation of integration tests for each gateway implementation

File: packages/contracts/package.json [hunk 2]
```diff
@@ -4,7 +4,8 @@
   "scripts": {
     "test": "forge test -vvv",
     "test:watch": "forge test -vvv --watch",
-    "start": "anvil & copyfiles ../../.env contracts/.env && forge script script/PublicResolver.s.sol --rpc-url http://localhost:8545 --broadcast -vv",
+    "start:l1": "anvil & copyfiles ../../.env contracts/.env && forge script script/PublicResolver.s.sol --rpc-url http://localhost:8545 --broadcast -vv",
+    "start:db": "anvil & copyfiles ../../.env contracts/.env && forge script script/DatabaseResolver.s.sol --rpc-url http://localhost:8545 --broadcast -vv",
```
Comment (pikonha) [inline packages/contracts/package.json:8]: the scripts got decoupled based on the storage implementation (L1, DB...)

File: packages/contracts/script/DatabaseResolver.s.sol
Comment (pikonha) [inline packages/contracts/script/DatabaseResolver.s.sol:1]: renamed to `DatabaseResolver` to enable the creation of different types of resolvers based on storage

File: packages/gateway/package.json [hunk 3]
```diff
@@ -7,7 +7,6 @@
     "test:watch": "vitest",
     "start": "dotenvx run --env-file=../../.env -- ts-node ./src/index.ts",
     "lint": "eslint . --ext .ts --fix",
-    "build": "tsc --build",
```
Comment (pikonha) [inline packages/gateway/package.json:10]: no need for this build command for now

File: packages/gateway/src/handlers/query.ts [hunk 4]
```diff
@@ -26,7 +26,7 @@ export function withQuery(): ccip.HandlerDescription {
           responses.push(result)
           failures.push(false)
         } catch (err) {
-          responses.push('0x0')
+          responses.push(pad('0x'))
```
Comment (pikonha) [inline packages/gateway/src/handlers/query.ts:29]: required for when the `query` call fails

File: packages/client/test/db.e2e.spec.ts [hunk 5]
```diff
@@ -0,0 +1,243 @@
...
+}
+
+describe('DatabaseResolver', () => {
+  let repo: InMemoryRepository
+  const rawNode = 'database.eth'
+  const node = namehash(rawNode)
+  const domains = new Map()
+  const domain = {
+    node,
+    ttl: 99712622115,
```
Comment (LeonardoVieira1630) [inline packages/client/test/db.e2e.spec.ts:132]: Does this value have something special? Why this number?

File: packages/client/tsconfig.json [hunk 6]
```diff
@@ -2,5 +2,7 @@
   "extends": "@blockful/tsconfig/tsconfig.json",
   "compilerOptions": {
     "resolveJsonModule": true,
+    "emitDecoratorMetadata": true,
+    "experimentalDecorators": true,
```
Comment (LeonardoVieira1630) [inline packages/client/tsconfig.json:6]: What is this about?

**Review by LeonardoVieira1630 — COMMENTED** [review]
Hi Lucas. I reviewed the Pull Request and have no improvement points to request. Good job. Had only two questions that i want you to answer, but no big deal.

File: packages/client/test/db.e2e.spec.ts (hunk 5 above)
Comment (pikonha) [reply packages/client/test/db.e2e.spec.ts:132]: just a number that the local blockchain will never reach

File: packages/client/tsconfig.json (hunk 6 above)
Comment (pikonha) [reply packages/client/tsconfig.json:6]: it is required when importing the entities from Typeorm due to the decorator syntax

**Pull request merged**